File: src/main/java/com/marand/auditer/demo/AuditController.java
package com.marand.auditer.demo;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.jms.core.JmsTemplate;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;


/**
 * @author Nejc Korasa
 */

@RestController
@RequestMapping("/audit")
public class AuditController
{
  private final JmsTemplate jmsTemplate;

  @Autowired
  public AuditController(final JmsTemplate jmsTemplate)
  {
    this.jmsTemplate = jmsTemplate;
  }

  @PostMapping
  public void send(@RequestBody final AuditDetails auditDetails)
  {
    System.out.println("Sending a transaction.");
    jmsTemplate.convertAndSend("auditQueue", auditDetails);
  }
}

File: src/main/kotlin/com/marand/auditor/rest/AuditController.kt
package com.marand.auditor.rest

import com.marand.auditor.AuditService
import com.marand.auditor.dto.AuditInfo
import io.swagger.annotations.ApiOperation
import io.swagger.annotations.ApiParam
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.jms.core.JmsTemplate
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController
import javax.validation.Valid


/**
 * @author Nejc Korasa
 */

@RestController
@RequestMapping("/audit")
open class AuditController @Autowired
constructor(private val jmsTemplate: JmsTemplate, private val auditService: AuditService) {

    @PostMapping("/mq")
    @ApiOperation(value = "Post Audit Info - add to Message queue")
    open fun auditMQ(@ApiParam(name = "auditInfo", required = true) @Valid @RequestBody auditInfo: AuditInfo) {

        jmsTemplate.convertAndSend("auditQueue", auditInfo)
    }

    @PostMapping("/direct")
    @ApiOperation(value = "Post Multiple Audit Info - save directly")
    open fun auditDirect(@ApiParam(name = "auditInfo", required = true) @Valid @RequestBody auditInfos: List<AuditInfo>) {

        auditService.auditBatch(auditInfos)
    }
}
